src/handle_finalize.c

```diff
@@ -1,10 +1,30 @@
 #include "figment_plugin.h"
 
+static bool withdrawal_address_matches_sender(ethPluginFinalize_t *msg) {
+    context_t *context = (context_t *) msg->pluginContext;
+
+    return memcmp(context->withdrawal_address, msg->address, ADDRESS_LENGTH) == 0;
+}
+
 void handle_finalize(void *parameters) {
     ethPluginFinalize_t *msg = (ethPluginFinalize_t *) parameters;
+    context_t *context = (context_t *) msg->pluginContext;
 
     msg->uiType = ETH_UI_TYPE_GENERIC;
-
     msg->numScreens = 1;
+
+    // Copy the withdrawal address (if present)
+    if (context->withdrawal_credentials[0] == ETH1_ADDRESS_WITHDRAWAL_PREFIX) {
```

Comment:
Do we need to also check the length of `context->withdrawal_credentials` to ensure it's not just a single 0x01 byte?

Reply:
Seems like we do the length check in the params handler. So, nvm.

Reply:
Yeah, `context->withdrawal_credentials` is always 32 bytes long and we check the length of the input parameter too ([here](https://github.com/figment-networks/ledger-app-plugin/blob/538fdfea295c60fb0e93232c8fb2f2ce59982819/src/handle_provide_parameter.c#L45-L49)).